sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AdaptiveSparkPlanExec.scala

```diff
@@ -194,16 +212,16 @@ case class AdaptiveSparkPlanExec(
   }
 
   private def applyQueryPostPlannerStrategyRules(plan: SparkPlan): SparkPlan = {
-    applyPhysicalRules(
+    applyPhysicalRulesWithRuleContext(
       plan,
       context.session.sessionState.adaptiveRulesHolder.queryPostPlannerStrategyRules,
-      Some((planChangeLogger, "AQE Query Post Planner Strategy Rules"))
-    )
+      Some((planChangeLogger, "AQE Query Post Planner Strategy Rules")))
   }
 
   @transient val initialPlan = context.session.withActive {
-    applyPhysicalRules(
-      applyQueryPostPlannerStrategyRules(inputPlan),
+    val planWithPostPlannerStrategy = applyQueryPostPlannerStrategyRules(inputPlan)
+    applyPhysicalRulesWithRuleContext(
+      planWithPostPlannerStrategy,
```

Comment:
Shall we change like the following simply by avoiding one-time used variable, `val planWithPostPlannerStrategy`?
```scala
- applyPhysicalRules(
+ applyPhysicalRulesWithRuleContext(
```

Reply:
It is required. We need to propagate the configs to the next rules, so we should make code like:
```
val x = withContext {}
val y = withContext {}
```
instead of 
```
val x = withContext {
  withContext {}
}
```